### DISCUSSION TITLE
New platform questions regarding IMU and stabilizer loop

### DISCUSSION THREAD

**Discussion by puliceworx** [bitcraze/discussions#830] [Q&A] [answered]
After digging through some very good discussions here I just cannot satisfy my issue.  

Stabilizer loop and IMU interrupts: 

I am using the exact same IMU (BMI088).  The only real difference I see is that cf 2.1 code appears to be using an older version of the Bosch bmi088 driver.  (Other than different MCU and different OS as well lol). 

I cannot for the life of me get the stabilizer loop warning to be any higher than a reported 975-978 rate range.  
This is bench testing so I ignore the rate warning displayed and log anytime its out of the range.  

So the question I have is mainly this, "Is a loop rate\imu frame rate of 975 good enough?" or do I really need to spend time figuring out why I cannot achieve 1K frame rate?

**Comment by krichardsson** [comment, marked as answer]
Hi,
If I remember correctly the stabilizer loop is driven by the interrupt from the gyro that is supposed to run at 1000Hz. Not sure why you get 975? Either the interrupt is not running at 1000Hz or the stabilizer task can not keep up?

> So the question I have is mainly this, "Is a loop rate\imu frame rate of 975 good enough?" or do I really need to spend time figuring out why I cannot achieve 1K frame rate?

My guess is that it will work any way, you can try :-) 
There will be some assumptions that will not hold in the code though, like time deltas that might be set to constants (at least in the PID controller I think). It will be pretty close though and I think it will fly. 
On the other hand, not understanding why you don't get 1000Hz might indicate that something else is off/weird. In my experience these little oddities tend to come back and bite you later.

**Comment by puliceworx** [comment]
I am very happy and very excited to report a perfect hover with my PAA3905 and stabilizer rate (IMU irq) of 975.  
Still more testing is needed to ensure all of this over time but I've got it.  

I tested the auto mode switching of the PAA3905 bright, low, and super.  Still maintains hover. 
My issues were the PAA3905.  To get it fixed I use the default startup orientation. 
No transposing of flowDelta X and Y and no negation of the same. 

I think I have fallen into a very fun hobby with this drone stuff lol 
#zephyrRTOS

**Comment by krichardsson** [comment]
Great!
